indexing-service/src/main/java/org/apache/druid/indexing/seekablestream/supervisor/SeekableStreamSupervisor.java

```diff
@@ -2014,9 +2014,24 @@ private void resumeAllActivelyReadingTasks()
     }
 
     Map<String, ListenableFuture<Boolean>> tasksToResume = new HashMap<>();
+    if (activelyReadingTaskGroups.isEmpty()) {
+      return;
+    }
+    // Resume only running tasks and not pending / waiting ones.
+    if (!taskMaster.getTaskRunner().isPresent()) {
+      return;
+    }
+    Set<String> runningTaskIds = taskMaster.getTaskRunner()
+                                           .get()
+                                           .getRunningTasks()
```

Comment:
Does this also return paused tasks? According to the javadoc of this method, this method is meant to resume paused tasks (and try to resume running tasks, which should be a no-op).

Reply:
Yes. Paused is an unrelated internal task state and the task runner considers such tasks to be running as well

Reply:
Just catching up here, isn't the issue that we kill off RUNNING tasks which are actively processing data/publishing segments? Or were we killing off WAITING tasks?

IIUC, the existing flow wouldn't affect PENDING or WAITING tasks anyway, as they haven't started execution yet.

Reply:
The code introduced in #13223 tried to resume any task in the set of `activelyReadingTaskGroups` irrespective of the task runner worker status and killed the task if it failed to respond with a successful status.

The issue was that there were PENDING / WAITING tasks in the set which couldn't respond to this request as they hadn't even begun RUNNING, and were killed

Reply:
Makes sense.

Have we seen a specific drawback of killing off PENDING/WAITING tasks or a race condition that may have adverse effects, or is this PR just a safety measure to avoid unnecessary operations?

Reply:
Killing pending tasks of a supervisor could lead to increased lag as it must re-create these tasks in the next run.
The failure could also be misleading as one cannot resume a task that hasn't begun running